cirq/sim/simulator.py

```diff
@@ -525,3 +530,43 @@ def dirac_notation(self, decimals=2):
             A pretty string consisting of a sum of computational basis kets
             and non-zero floats of the specified accuracy."""
         return wave_function.dirac_notation(self.state, decimals)
+
+    def density_matrix(self, indices=None):
+        """Returns the density matrix of the wavefunction.
+
+        Calculate the density matrix for the system on the given qubit
+        indices, with the qubits not in indices that are present in self.state
+        traced out. If indices is None the full density matrix for self.state
+        is returned.
+        For example:
+            self.state = np.array([1/np.sqrt(2), 1/np.sqrt(2)],
+                dtype=np.complex64)
+            indices = None
+            gives us \rho = \begin{bmatrix}
+                                0.5 & 0.5
+                                0.5 & 0.5
+                            \end{bmatrix}
+
```

Comment:
should document that it follows the np.kron order in relation to order of indices.

Reply:
fixed